src/util/metadata/side_metadata/global.rs

```diff
@@ -178,31 +224,21 @@ impl SideMetadataSpec {
                 - start.align_down(BYTES_IN_CHUNK))
                 / BYTES_IN_CHUNK;
             if chunk_num == 0 {
-                memory::zero(
-                    meta_start,
-                    address_to_meta_address(self, start + size) - meta_start,
-                );
+                zero(start, start + size);
             } else {
                 let second_data_chunk = start.align_up(BYTES_IN_CHUNK);
                 // bzero the first sub-chunk
-                memory::zero(
-                    meta_start,
-                    address_to_meta_address(self, second_data_chunk) - meta_start,
-                );
+                zero(start, second_data_chunk);
```

Comment:
Two problems:

1.  If `start` is already aligned to `BYTES_IN_CHUNK`, then `start.align_up(BYTES_IN_CHUNK)` will be equal to `start`, and it will end up calling `zero(start, start)`.
2.  If the metadata is discontiguous, passing `second_data_chunk` to the `zero` function will result in `meta_end` being in a different metadata chunk, and unrelated metadata in between will be overwritten.

Reply:
1. I added a check, if the size to be zeroed is 0, then the method simply returns.
2. See my reply for the above comments.